staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -136,10 +126,19 @@ type SharedInformer interface {
 	// between different handlers.
 	AddEventHandler(handler ResourceEventHandler)
 	// AddEventHandlerWithResyncPeriod adds an event handler to the
-	// shared informer using the specified resync period.  The resync
-	// operation consists of delivering to the handler an update
+	// shared informer with the requested resync period; zero means
+	// this handler does not care about resyncs.  The resync operation
+	// consists of delivering to the handler an add or update
```

Comment:
> The resync operation consists of delivering to the handler an add or update

Isn't it impossible for a Resync to trigger an Add notification?
An Add notification si triggered by a Delta whose type is not `Deleted` and **whose object is not in the SharedInformer's cache**. But by its nature `Resync()` only adds a Delta for an object if that object is already in the SharedInformer's cache. The case where a Delta added by a `Resync()` is preceded by a `Deleted` Delta is impossible, because `Resync()` does not enqueue a Delta for an object it is considering if the DeltaFIFO already holds a Delta for that object.

Reply:
Thanks for catching that.  Fixed in the latest rev.